apps/google-docs/functions/agents/documentParser.agent.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * INTEG-3263: Content Type Parser Agent
+ *
+ * Simple agent that takes JSON input and sends it to OpenAI/ChatGPT
+ * to generate Contentful content type definitions.
+ * See https://contentful.atlassian.net/wiki/spaces/ECO/pages/5850955777/RFC+Google+Docs+V1+AI-Gen
+ * for more details.
+ */
+
+import { openai } from '@ai-sdk/openai';
+import { generateText } from 'ai';
+
+/**
+ * Configuration for the content type parser
+ */
+interface ContentTypeParserConfig {
+  openaiApiKey: string;
+  modelVersion: string;
+  jsonData: any;
+  document: any;
+}
```

Comment:
Should this type be named `DocumentParserConfig` instead? The content type parser function already has an interface with this name.